programs/client/Client.cpp

```diff
@@ -512,6 +512,7 @@ catch (...)
 
 void Client::connect()
 {
+    config().setString("host", hosts[0]);
```

Comment:
This line shouldn't be here as soon as you remove field `hosts` from `Client` class

Reply:
It is a crutch, which I will move to initialization (more you can see in conversation below https://github.com/ManagedDatabases/ClickHouse/pull/4#discussion_r740562604)

Reply:
Well, so as soon as you don't want to add hosts to config, you just don't need to set it in the beginning. You should either set it in config or don't set at all. I see you want to save hosts anywhere, but please write clean code